### ISSUE TITLE
User registration, email verification and password reset

### ISSUE THREAD

**Issue by dennis95stumm** [THM-Health/PILOS#71]
**Describe the solution you'd like**
It should be possible to activate user registration trough a application setting. Also the database users should have the possibility to reset their password and a configurable email verification should be implemented.

**Comment by SamuelWei** [comment]
The admin be should choose between random password and a custom one for users that are created via the admin interface. If the password is random, send welcome email with a password (re)set link to the new user.

**Comment by SamuelWei** [comment]
> The admin be should choose between random password and a custom one for users that are created via the admin interface. If the password is random, send welcome email with a password (re)set link to the new user.

@dennis95stumm I'm not sure if #70 is required if we can manually create user and send them an invitation email with the ability to set the password. Creating users via the interface as admin isn't possible with greenlight (at least to my knowledge)

**Comment by dennis95stumm** [comment]
@SamuelWei This ticket was about self registration and self password reset and not about the admin who changes the users password! But it could also be implemented here, then the ticket #70 would be useless but in the [Requirements](https://github.com/THM-Health/PILOS/wiki/Anforderungen) the third point is something like the describes user invitation.

**Comment by SamuelWei** [comment]
@dennis95stumm I agree. Maybe we would split this. Having a self registration feature is definitely important, but nothing that we need to start using this software. For the first release we would need a way for the admin to create non-ldap users (already possibile) and a way to reset passwords for non-ldap users. It would be great if we could add a feature that non-ldap users get a welcome email and can set their password through a link in the email. 

If you think that would be better addresses in a new issue, you can open a new one.

**Comment by dennis95stumm** [comment]
Okay I will create a new issue since this isn't the right place for the issue!

**Comment by ashishkpaul** [comment]
hello sir, I am missing self user registration and password reset feature in pilos. It would be great if you could add this feature or could connect with [keycloak](https://www.keycloak.org/)

**Comment by SamuelWei** [comment]
@ashishkpaul We plan to add OpenID-Connect and SAML 2.0 support, so you can connect PILOS with keycloak to allow self registration

**Comment by ashishkpaul** [comment]
Thanks for informing sir, Will host again

**Comment by samuelwei** [comment]
@ashishkpaul OIDC was added in PILOS v4.8.0
